sdk/security/keyvault/internal/challenge_policy.go

```diff
@@ -105,29 +104,11 @@ func parseTenant(url string) string {
 	return tenant
 }
 
-type challengePolicyError struct {
```

Comment:
So this was just redundant?

Reply:
Yes. I just edited an explanation into the top comment but the gist is, this type existed to make sure the policy returns non retriable errors. BearerTokenPolicy does that in the new azcore, so we don't need this type anymore.